src/opening-period/form/OpeningPeriodForm.tsx

```diff
@@ -0,0 +1,254 @@
+} from './form-helpers/form-helpers';
+import OpeningPeriodDescription from '../description/OpeningPeriodDescription';
+import TimeSpan from '../time-span/TimeSpan';
+import './OpeningPeriodForm.scss';
+
+interface OpeningPeriodFormData {
+  openingPeriodTitle: string;
+  openingPeriodOptionalDescription: string;
+  openingPeriodBeginDate: string | undefined;
+  openingPeriodEndDate: string | undefined;
+  timeSpans: Array<TimeSpanFormFormat> | {}[];
+}
+
+type SubmitStatus = 'init' | 'succeeded' | 'error';
+
+type NotificationTexts = {
+  label: string;
+  text: string;
+};
+
+const formatResourceStateOptionsToFormFormat = (
+  resourceStateApiOptions: ResourceStateApiOption[]
+): ResourceStateOption[] =>
+  resourceStateApiOptions.map((optionInApiFormat: ResourceStateApiOption): {
+    value: string;
+    label: string;
+  } => {
+    return {
+      value: optionInApiFormat.value,
+      label: `${
+        typeof optionInApiFormat.display_name === 'string'
+          ? optionInApiFormat.display_name
+          : optionInApiFormat.display_name.fi
+      }`,
+    };
+  });
+
+export default function OpeningPeriodForm({
+  formId,
+  datePeriod,
+  resourceId,
+  datePeriodOptions,
+  submitFn,
+  successTextAndLabel,
+  errorTextAndLabel,
+}: {
+  formId: string;
+  datePeriod?: DatePeriod;
+  resourceId: number;
+  datePeriodOptions: DatePeriodOptions;
+  submitFn: (datePeriod: DatePeriod) => Promise<DatePeriod>;
+  successTextAndLabel: NotificationTexts;
+  errorTextAndLabel: NotificationTexts;
+}): JSX.Element {
+  const language = Language.FI;
+
+  const resourceStateOptionsInApiFormat: ResourceStateApiOption[] =
+    datePeriodOptions.actions.POST.resource_state.choices;
+
+  const resourceStateOptions: ResourceStateOption[] = formatResourceStateOptionsToFormFormat(
+    resourceStateOptionsInApiFormat
+  );
+
+  const spans: TimeSpanFormFormat[] | {}[] =
+    datePeriod?.time_span_groups[0] &&
+    datePeriod?.time_span_groups[0].time_spans
+      ? formatApiTimeSpansToFormFormat(
+          datePeriod?.time_span_groups[0].time_spans
+        )
+      : [{}];
+
+  const [periodBeginDate, setPeriodBeginDate] = useState<Date | null>(
+    datePeriod?.start_date ? new Date(datePeriod?.start_date) : null
+  );
+  const [periodEndDate, setPeriodEndDate] = useState<Date | null>(
+    datePeriod?.end_date ? new Date(datePeriod?.end_date) : null
+  );
+
+  const formValues: OpeningPeriodFormData = {
+    openingPeriodTitle: datePeriod?.name[language] || '',
+    openingPeriodOptionalDescription: datePeriod?.description[language] || '',
+    openingPeriodBeginDate: datePeriod?.start_date,
+    openingPeriodEndDate: datePeriod?.end_date,
+    timeSpans: spans,
+  };
+
+  const { register, handleSubmit, errors, control, setValue } = useForm<
+    OpeningPeriodFormData
+  >({
+    mode: 'all',
+    defaultValues: formValues,
+  });
+  const { fields, append, remove } = useFieldArray({
+    control,
+    keyName: 'timeSpanUiId',
```

Comment:
Is this required? Didn't seem to find it from code hmm..

Reply:
React-hook logged a warning because it uses id as it’s own internal prop. It  conflicted with timespan’s own id (in edit mode)